core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStore.java

```diff
@@ -151,6 +183,15 @@ public void init() throws StoreInitializationException {
     }
   }
 
+  // helper method to only add a property if its not the empty string. This is required
+  // because some Kafka client configs expect a null default value, yet ConfigDef doesn't
+  // support null default values.
+  public static void putIfNotEmptyString(String parameter, String value, Properties props) {
```

Comment:
@Ishiihara, I've been experimenting with this, and unless I'm missing something, the changes in that PR don't solve my issue. My issue is that I want a null default value, not an empty string default value.

I modified `ConfigDef` to support my use case. Take a look: https://github.com/alexlod/common/commit/eea951b01653ab4abfc89d262e870392ed7b8ed2

I'm not comfortable committing the changes as they stand, because I noticed some odd behavior in my `testNullDefaultValueWithValidator()` test. Namely, if a "null" validator is used, the test passes. However, if a validator is not used, the test fails, because the actual value is `""`. I expect I could fix this, but I wonder if it's worth it considering the `ConfigDef` changes are non-trivial.

Anyway, take a look, and let me know what you think.